server/http.go

```diff
@@ -84,84 +84,63 @@ func (p *Plugin) completeUserOAuthToZoom(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	ctx := context.Background()
-	conf, err := p.getOAuthConfig()
-	if err != nil {
-		http.Error(w, "error in oauth config", http.StatusInternalServerError)
-	}
-
 	code := r.URL.Query().Get("code")
 	if len(code) == 0 {
 		http.Error(w, "missing authorization code", http.StatusBadRequest)
 		return
 	}
 
-	state := r.URL.Query().Get("state")
-	stateComponents := strings.Split(state, "_")
-
-	if len(stateComponents) != zoomStateLength {
-		log.Printf("stateComponents: %v, state: %v", stateComponents, state)
-		http.Error(w, "invalid state", http.StatusBadRequest)
-	}
-	key := fmt.Sprintf("%v_%v", stateComponents[0], stateComponents[1])
-
-	var storedState []byte
-	var appErr *model.AppError
-	storedState, appErr = p.API.KVGet(key)
+	storedState, appErr := p.fetchOAuthUserState(authedUserID)
 	if appErr != nil {
-		fmt.Println(appErr)
-		http.Error(w, "missing stored state", http.StatusBadRequest)
+		http.Error(w, "missing stored state", http.StatusNotFound)
 		return
 	}
 
-	if string(storedState) != state {
-		http.Error(w, "invalid state", http.StatusBadRequest)
+	userID, channelID, err := parseOAuthUserState(storedState)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
-	userID := stateComponents[1]
-	channelID := stateComponents[2]
-
-	appErr = p.API.KVDelete(state)
-	if appErr != nil {
-		p.API.LogWarn("failed to delete state from db", "error", appErr.Error())
-	}
```

Comment:
No need to explicitly delete anymore because we're now setting a 5 min TTL when storing the state.

Reply:
It is better to delete it, even if it will expire at some point.